### PR TITLE
DD Name Unit Test

### PR DESCRIPTION
**Pull request by ianna**
* Add a unit test
* Fix a bug in DDName

Note, it is based on https://github.com/cms-sw/cmssw/pull/24263 and https://github.com/cms-sw/cmssw/pull/24313

@Dr15Jones and @cvuosalo - I'm not sure what was an original requirement behind a DDName design. I think, the unit test demonstrates one usecase: the registry keeps only unique names.

### CODE REVIEW

**Comment by ianna** [comment]
please test

**Comment by ianna** [comment]
please test

**Comment by andrius-k** [comment]
+1
A minor change for DQM

File: DetectorDescription/Core/test/DDName.cppunit.cc [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+CPPUNIT_TEST_SUITE_REGISTRATION(testDDName);
+
+void
+testDDName::buildIt() {
+  DDName name;
+  DDName anotherName;
+  std::cerr << "\nDDName name=" << name << " id=" << name.id()
+	    << ", another name=" << anotherName << " id=" << anotherName.id();
+  int a[100000]{};
```
Comment (Dr15Jones) [inline DetectorDescription/Core/test/DDName.cppunit.cc]: This should just be uninitialized values. Is that what you want?

Comment (Dr15Jones) [reply DetectorDescription/Core/test/DDName.cppunit.cc]: Doing
```
int a[100000] = {};
```
Will zero initialize all the elements.

Comment (Dr15Jones) [reply DetectorDescription/Core/test/DDName.cppunit.cc]: I misread one of the web sites info. Testing using 'compiler explorer' does show that what you did will initialize all values to 0.  Sorry for the incorrect comment.

**Comment by ianna** [comment]
+1

**Comment by alja** [comment]
+1

**Comment by kpedro88** [comment]
+upgrade

**Comment by kpedro88** [comment]
+1

**Pull request merged**